.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,68 @@
+# Please reference work here https://github.com/EVerest/everest-core/tree/main/.github/workflows
+# TODO: modify to reuse the above workflow to DRY up CI.
+
+name: Build and test liblog
+on: 
+  push:
+    branch: "**"
+  pull_request: {}
```

Comment:
you are running this on all pushes but no pull requests?
I believe that `pull_request: {}` means no branches are specified for the pull request, which means that it won't ever run on PR. Have you tested this with a pull request?

Reply:
Hm. I guess if no branch is specified, then the event matches any branch
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-a-single-event

But I still haven't found documentation indicating that an explicitly specified blank list is the same as "no branch"

Reply:
Cleaned this up so that it's clearer. I've used this pattern when I want runs on all branches and PRs:
```
on:
  push:
  pull_request:
```
Confirmed that it's run here.